pkg/migration/templates/kubeadm_controlplane_azure.yaml.tmpl

```diff
@@ -0,0 +1,286 @@
+      owner: root:root
+    - contentFrom:
+        secret:
+          key: tls.key
+          name: {{.ClusterID}}-ca
+      owner: root:root
+      path: /etc/kubernetes/pki/etcd/ca.key
+      permissions: "0600"
+    - contentFrom:
+        secret:
+          key: tls.crt
+          name: {{.ClusterID}}-ca
+      owner: root:root
+      path: /etc/kubernetes/pki/ca.crt
+      permissions: "0640"
+    - contentFrom:
+        secret:
+          key: tls.key
+          name: {{.ClusterID}}-ca
+      owner: root:root
+      path: /etc/kubernetes/pki/ca.key
+      permissions: "0600"
+    - contentFrom:
+        secret:
+          name: {{.ClusterID}}-etcd
+          key: "tls.crt"
+      owner: root:root
+      path: /etc/kubernetes/pki/etcd/old-etcd-cert.pem
+      permissions: "0640"
+    - contentFrom:
+        secret:
+          name: {{.ClusterID}}-etcd
+          key: "tls.key"
+      owner: root:root
+      path: /etc/kubernetes/pki/etcd/old-etcd-key.pem
+      permissions: "0640"
+    - contentFrom:
+        secret:
+          name: {{.ClusterID}}-service-account
+          key: "tls.crt"
+      owner: root:root
+      path: /etc/kubernetes/pki/sa.pub
+      permissions: "0640"
+    - contentFrom:
+        secret:
+          name: {{.ClusterID}}-service-account
+          key: "tls.key"
+      owner: root:root
+      path: /etc/kubernetes/pki/sa.key
+      permissions: "0640"
+    - contentFrom:
+        secret:
+          name: {{.ClusterID}}-k8s-encryption-config
+          key: encryption
+      owner: root:root
+      path: "/etc/kubernetes/encryption/k8s-encryption-config.yaml"
+      permissions: "0644"
+    - contentFrom:
+        secret:
+          name: {{.ClusterID}}-proxy-config
+          key: proxy
+      owner: root:root
+      path: "/etc/kubernetes/config/proxy-config.yml"
+      permissions: "0644"
+    - contentFrom:
+        secret:
+          name: {{.ClusterID}}-kubeconfig
+          key: value
+      owner: root:root
+      path: "/etc/kubernetes/config/proxy-kubeconfig.yaml"
+      permissions: "0644"
+    - content: |
+        #!/bin/sh
+        # create etcd ca bundle
+        cat /etc/kubernetes/pki/etcd/ca.crt /etc/kubernetes/pki/etcd/old-etcd-ca.pem > /etc/kubernetes/pki/etcd/ca-bundle.pem
+        # get ETCDCTL
+        DOWNLOAD_URL=https://github.com/etcd-io/etcd/releases/download
+        ETCD_VER={{.EtcdVersion}}
+        rm -f /tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz
+        rm -rf /tmp/etcd && mkdir -p /tmp/etcd
+        curl -L ${DOWNLOAD_URL}/${ETCD_VER}/etcd-${ETCD_VER}-linux-amd64.tar.gz -o /tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz
+        tar xzvf /tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz -C /tmp/etcd --strip-components=1
+        rm -f /tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz
+        /tmp/etcd/etcdctl version
+        # add hosts entry to reach etcd on private address
+        echo "" >>/etc/hosts
+        echo "{{.ClusterMasterIP}} etcd.{{.ClusterID}}.k8s.{{.InstallationBaseDomain}}" >>/etc/hosts
```

Comment:
Shouldn't that be the LB address? Why do we have to configure that localy?

Reply:
Ping @whites11 - I believe this was some kind of workaround for that kubeadm join logic to work.

Reply:
it's a firewall issue. The etcd load balancer is a public one and the DNS entry resolves to a public IP.
the firewall rules restrict access to such public IP to some addresses only and therefore the new master nodes do not get access through the load balancer.
to circumvent that we need to reach etcd on the private IP address of the old master node.
This is a workaround that will eventually go away, but it's the easiest strategy for now

Reply:
> will eventually go away

Do we need to track that somewhere?

Reply:
> Do we need to track that somewhere?

I made a ticket for that: https://github.com/giantswarm/giantswarm/issues/16443